src/Agent.Worker/TestResults/TestDataPublisher.cs

```diff
@@ -57,6 +57,7 @@ public void InitializePublisher(IExecutionContext context, string projectName, V
             _testResultsServer.InitializeServer(connection, _executionContext);
             var extensionManager = HostContext.GetService<IExtensionManager>();
             _featureFlagService = HostContext.GetService<IFeatureFlagService>();
+            _featureFlagService.InitializeFeatureService(_executionContext, connection);
```

Comment:
Just for clarification - do we need to handle the _featureFlagService null scenario?, I mean is it possible?

Reply:
Taken care by the GetService call : 
![image](https://github.com/microsoft/azure-pipelines-agent/assets/110218555/1953887e-e896-4db9-8334-6d7736cef742)